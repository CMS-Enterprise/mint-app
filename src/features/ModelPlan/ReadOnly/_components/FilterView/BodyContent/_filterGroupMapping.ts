import { ModelViewFilter, MtoCommonSolutionKey } from 'gql/generated/graphql';

import { getKeys } from 'types/translation';

// Importing ModelViewFilter enum and converting to lowercase to work easily with FE routes
export const filterGroups = getKeys(ModelViewFilter).map(filter =>
  filter.toLowerCase()
);

export type FilterGroup = (typeof filterGroups)[number];

// Map for url route params to ModelViewFilter enum
export const filterGroupKey: Record<FilterGroup, ModelViewFilter> = {
  cbosc: ModelViewFilter.CBOSC,
  ccw: ModelViewFilter.CCW,
  cmmi: ModelViewFilter.CMMI,
  dfsdm: ModelViewFilter.DFSDM,
  iddoc: ModelViewFilter.IDDOC,
  ipc: ModelViewFilter.IPC,
  mdm: ModelViewFilter.MDM,
  oact: ModelViewFilter.OACT,
  pbg: ModelViewFilter.PBG
};

// Map of operational solutions that should be rendered on each filter view
export const filteredGroupSolutions: Record<
  FilterGroup,
  MtoCommonSolutionKey[]
> = {
  cbosc: [MtoCommonSolutionKey.CBOSC],
  ccw: [MtoCommonSolutionKey.CCW, MtoCommonSolutionKey.SHARED_SYSTEMS],
  dfsdm: [MtoCommonSolutionKey.IPC],
  ipc: [
    MtoCommonSolutionKey.INNOVATION,
    MtoCommonSolutionKey.ACO_OS,
    MtoCommonSolutionKey.CDX,
    MtoCommonSolutionKey.IPC
  ],
  iddoc: [MtoCommonSolutionKey.INNOVATION, MtoCommonSolutionKey.ACO_OS],
  mdm: [MtoCommonSolutionKey.MDM_POR],
  pbg: [
    MtoCommonSolutionKey.IPC,
    MtoCommonSolutionKey.SHARED_SYSTEMS,
    MtoCommonSolutionKey.EFT
  ]
};
