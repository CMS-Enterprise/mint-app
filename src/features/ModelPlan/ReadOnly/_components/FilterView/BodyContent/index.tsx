import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Alert, Grid, Link } from '@trussworks/react-uswds';
import ReadOnlyBeneficiaries from 'features/ModelPlan/ReadOnly/Beneficiaries';
import ReadOnlyCRTDLs from 'features/ModelPlan/ReadOnly/CRTDLs';
import ReadOnlyDataExchangeApproach from 'features/ModelPlan/ReadOnly/DataExchangeapproach';
import ReadOnlyGeneralCharacteristics from 'features/ModelPlan/ReadOnly/GeneralCharacteristics';
import ReadOnlyModelBasics from 'features/ModelPlan/ReadOnly/ModelBasics';
import ReadOnlyOpsEvalAndLearning from 'features/ModelPlan/ReadOnly/OpsEvalAndLearning';
import ReadOnlyParticipantsAndProviders from 'features/ModelPlan/ReadOnly/ParticipantsAndProviders';
import ReadOnlyPayments from 'features/ModelPlan/ReadOnly/Payments';
import ReadOnlyTeamInfo from 'features/ModelPlan/ReadOnly/Team';
import OperationalNeedsTable from 'features/ModelPlan/TaskList/ITSolutions/Home/operationalNeedsTable';

import Divider from 'components/Divider';
import usePlanTranslation from 'hooks/usePlanTranslation';
import {
  getKeys,
  TranslationFieldProperties,
  TranslationPlan
} from 'types/translation';

import {
  filteredGroupSolutions,
  filterGroupKey,
  filterGroups
} from './_filterGroupMapping';

const components: Record<string, React.ElementType> = {
  basics: ReadOnlyModelBasics,
  generalCharacteristics: ReadOnlyGeneralCharacteristics,
  participantsAndProviders: ReadOnlyParticipantsAndProviders,
  beneficiaries: ReadOnlyBeneficiaries,
  opsEvalAndLearning: ReadOnlyOpsEvalAndLearning,
  payments: ReadOnlyPayments,
  dataExchangeApproach: ReadOnlyDataExchangeApproach
};

const FitleredViewSection = ({
  children,
  lastSection,
  sectionName
}: {
  children: React.ReactNode;
  lastSection?: boolean;
  sectionName: string;
}) => {
  return (
    <div
      className={`filtered-view-section filtered-view-section--${sectionName} margin-bottom-6 ${
        !lastSection
          ? 'border-bottom border-base-light padding-bottom-4'
          : 'padding-bottom-0'
      }`}
    >
      {children}
    </div>
  );
};

const BodyContent = ({
  modelID,
  filteredView
}: {
  modelID: string;
  filteredView: (typeof filterGroups)[number];
}) => {
  const { t } = useTranslation('filterView');
  const { t: opSolutionsMiscT } = useTranslation('opSolutionsMisc');

  const filterMappings = usePlanTranslation();

  const mappedQuestions = getAllFilterViewQuestions(
    filterMappings,
    filteredView
  );

  return (
    <Grid>
      <FitleredViewSection sectionName="model-team">
        <h2 className="margin-top-0 margin-bottom-4">{t('modelTeam')}</h2>
        <ReadOnlyTeamInfo
          modelID={modelID}
          isViewingFilteredView
          filteredView={filteredView}
        />
      </FitleredViewSection>

      {Object.keys(mappedQuestions).map(task => {
        const FitleredViewSectionContent = components[task];

        if (!FitleredViewSectionContent) return null;

        return (
          <FitleredViewSection
            sectionName={task}
            key={task}
            lastSection={task === Object.keys(mappedQuestions).pop()}
          >
            <FitleredViewSectionContent
              modelID={modelID}
              filteredView={filteredView}
            />
          </FitleredViewSection>
        );
      })}

      {/* Operational Solutions table for filtered views */}
      {filteredGroupSolutions[filteredView] && (
        <div>
          <Divider className="margin-top-8" />
          <h2 className="margin-top-5">
            {opSolutionsMiscT('headingReadOnly')}
          </h2>
          <OperationalNeedsTable
            className="margin-bottom-5"
            modelID={modelID}
            type="needs"
            readOnly
            hideGlobalFilter
            hiddenColumns={[opSolutionsMiscT('itSolutionsTable.actions')]}
            filterSolutions={filteredGroupSolutions[filteredView]}
          />
        </div>
      )}

      {/* CR and TDLs table for filtered views */}
      {filteredGroupSolutions[filteredView] && (
        <div>
          <Divider className="margin-y-8" />
          <ReadOnlyCRTDLs />
        </div>
      )}

      <div className="margin-top-4 padding-top-5 border-top-1px border-base-light">
        <Alert type="info" noIcon headingLevel="h4">
          <span className="margin-y-0 font-body-sm text-bold display-block">
            {t('alert.bodyContentHeading')}
          </span>
          <Trans i18nKey="filterView:alert.content">
            indexOne
            <Link href="mailto:MINTTeam@cms.hhs.gov">helpTextEmail</Link>
            indexTwo
          </Trans>
        </Alert>
      </div>
    </Grid>
  );
};

/*
  Traversed all translation mappings to return an refined mapping pertaining to the relative filter view
*/
export const getAllFilterViewQuestions = (
  filterMappings: TranslationPlan,
  filteredView: (typeof filterGroups)[number]
) => {
  let mappedQuestions: Record<string, string[]> = {};

  // Diving into each model plan section, as well as each question to identify if it contains the appropriate filter view mapping
  getKeys(filterMappings).forEach(section => {
    getKeys(filterMappings[section]).forEach(question => {
      const filterGroupConfig = (
        filterMappings[section][question] as TranslationFieldProperties
      )?.filterGroups;

      if (
        Array.isArray(filterGroupConfig) &&
        filterGroupConfig.includes(filterGroupKey[filteredView])
      ) {
        if (mappedQuestions[section as string]) {
          mappedQuestions[section as string].push(
            (filterMappings[section][question] as TranslationFieldProperties)
              .gqlField
          );
        } else {
          mappedQuestions[section as string] = [
            (filterMappings[section][question] as TranslationFieldProperties)
              .gqlField
          ];
        }
      }
    });
  });

  // Transferring mapped questions from modelPlan to basics. Readonly treats modelPlan as if were in basics schema
  if (mappedQuestions.modelPlan) {
    mappedQuestions = {
      basics: (mappedQuestions.basics || []).concat(mappedQuestions.modelPlan),
      ...mappedQuestions
    };
  }
  return mappedQuestions;
};

export default BodyContent;
