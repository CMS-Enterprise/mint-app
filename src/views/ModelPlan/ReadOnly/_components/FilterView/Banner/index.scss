@use 'viewports' as *;

.banner {
  flex-direction: column;
  gap: 1rem;

  @media screen and (min-width: $desktop) {
    flex-direction: row;
  }

  @media print {
    padding-left: 1rem;
  }
}
