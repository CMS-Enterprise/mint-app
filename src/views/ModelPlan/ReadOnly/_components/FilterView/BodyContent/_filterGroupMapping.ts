export const filterGroups = [
  'cmmi',
  'oact',
  'dfsdm',
  'ccw',
  'ipc',
  'iddoc',
  'pbg',
  'mdm',
  'cbosc'
] as const;

export type FilterGroup = typeof filterGroups[number];

const FilterGroupMap: Record<typeof filterGroups[number], any> = {
  cmmi: {
    'general-characteristics': [
      'alternativePaymentModelTypes',
      'alternativePaymentModelTypesNote',
      'keyCharacteristics',
      'keyCharacteristicsNote',
      'keyCharacteristicsOther',
      'participationOptions',
      'agreementTypes',
      'multiplePatricipationAgreementsNeeded',
      'authorityAllowances',
      'waiversRequired'
    ],
    'participants-and-providers': [
      'participants',
      'medicareProviderType',
      'statesEngagement',
      'selectionMethod'
    ],
    beneficiaries: [
      'beneficiarySelectionMethod',
      'beneficiarySelectionFrequency'
    ],
    'ops-eval-and-learning': [
      'benchmarkForPerformance',
      'computePerformanceScores',
      'riskAdjustPerformance',
      'riskAdjustPayments',
      'dataNeededForMonitoring',
      'dataToSendParticicipants',
      'shareCclfData',
      'developNewQualityMeasures',
      'qualityPerformanceImpactsPayment',
      'dataSharingFrequency',
      'dataCollectionFrequency'
    ],
    payments: [
      'payType',
      'payClaims',
      'nonClaimsPayments',
      'canParticipantsSelectBetweenPaymentMechanisms',
      'anticipatedPaymentFrequency',
      'willRecoverPayments'
    ]
  },
  oact: {
    basics: ['nameHistory'],
    'general-characteristics': [
      'alternativePaymentModelTypes',
      'alternativePaymentModelTypesNote'
    ],
    'participants-and-providers': [
      'providerAdditionFrequency',
      'providerAddMethod',
      'providerLeaveMethod'
    ],
    beneficiaries: ['precedenceRules'],
    payments: [
      'fundingSource',
      'fundingSourceTrustFund',
      'fundingSourceR',
      'fundingSourceRTrustFund',
      'payClaims',
      'isContractorAwareTestDataRequirements',
      'beneficiaryCostSharingLevelAndHandling',
      'waiveBeneficiaryCostSharingForAnyServices'
    ]
  },
  dfsdm: {
    basics: ['nameHistory', 'modelType', 'goal', 'performancePeriodStarts'],
    'participants-and-providers': [
      'expectedNumberOfParticipants',
      'estimateConfidence'
    ],
    payments: [
      'fundingSource',
      'fundingSourceTrustFund',
      'fundingSourceR',
      'numberPaymentsPerPayCycle',
      'planningToUseInnovationPaymentContractor',
      'anticipatedPaymentFrequency',
      'paymentStartDate'
    ]
  },
  ccw: {
    basics: ['nameHistory', 'performancePeriodStarts'],
    'participants-and-providers': [
      'expectedNumberOfParticipants',
      'estimateConfidence'
    ],
    'ops-eval-and-learning': [
      'ccmInvolvment',
      'sendFilesBetweenCcw',
      'appToSendFilesToKnown',
      'useCcwForFileDistribiutionToParticipants'
    ],
    payments: ['sharedSystemsInvolvedAdditionalClaimPayment']
  },
  ipc: {
    basics: [
      'nameHistory',
      'modelCategory',
      'cmsCenters',
      'cmmiGroups',
      'modelType',
      'goal',
      'completeICIP',
      'clearanceStarts',
      'clearanceEnds',
      'announced',
      'applicationsStart',
      'applicationsEnd',
      'performancePeriodStarts',
      'performancePeriodEnds',
      'wrapUpEnds'
    ],
    'general-characteristics': ['rulemakingRequired'],
    'participants-and-providers': [
      'participants',
      'medicareProviderType',
      'statesEngagement',
      'expectedNumberOfParticipants',
      'estimateConfidence',
      'providerAdditionFrequency',
      'providerAddMethod',
      'providerLeaveMethod'
    ],
    payments: [
      'fundingSource',
      'fundingSourceTrustFund',
      'fundingSourceR',
      'fundingSourceRTrustFund',
      'payType',
      'nonClaimsPayments',
      'numberPaymentsPerPayCycle',
      'planningToUseInnovationPaymentContractor',
      'anticipatedPaymentFrequency',
      'willRecoverPayments',
      'anticipateReconcilingPaymentsRetrospectively',
      'paymentStartDate'
    ]
  },
  iddoc: {
    basics: [
      'nameHistory',
      'modelType',
      'goal',
      'announced',
      'performancePeriodStarts',
      'phasedIn',
      'phasedInNote'
    ],
    'general-characteristics': [
      'keyCharacteristics',
      'keyCharacteristicsNote',
      'keyCharacteristicsOther',
      'geographiesTargeted',
      'geographiesTargetedTypes',
      'geographiesTargetedAppliedTo',
      'rulemakingRequired'
    ],
    'participants-and-providers': [
      'participants',
      'medicareProviderType',
      'statesEngagement',
      'modelApplicationLevel',
      'selectionMethod',
      'participantsIds',
      'providerOverlap',
      'providerOverlapHierarchy'
    ],
    beneficiaries: [
      'treatDualElligibleDifferent',
      'excludeCertainCharacteristics'
    ],
    'ops-eval-and-learning': [
      'contractorSupport',
      'contractorSupportHow',
      'iddocSupport',
      'technicalContactsIdentified',
      'captureParticipantInfo',
      'icdOwner',
      'draftIcdDueDate',
      'uatNeeds',
      'stcNeeds',
      'testingTimelines',
      'dataMonitoringFileTypes',
      'dataResponseType',
      'dataResponseFileFrequency',
      'dataFullTimeOrIncremental',
      'unsolicitedAdjustmentsIncluded',
      'produceBenefitEnhancementFiles',
      'fileNamingConventions',
      'dataNeededForMonitoring',
      'dataSharingStarts',
      'dataSharingFrequency',
      'dataCollectionStarts',
      'anticipatedChallenges'
    ],
    payments: [
      'shouldAnyProvidersExcludedFFSSystems',
      'changesMedicarePhysicianFeeSchedule',
      'affectsMedicareSecondaryPayerClaims',
      'payModelDifferentiation',
      'creatingDependenciesBetweenServices',
      'needsClaimsDataCollection',
      'providingThirdPartyFile',
      'isContractorAwareTestDataRequirements',
      'beneficiaryCostSharingLevelAndHandling',
      'waiveBeneficiaryCostSharingForAnyServices',
      'nonClaimsPayments',
      'waiverOnlyAppliesPartOfPayment',
      'planningToUseInnovationPaymentContractor',
      'anticipateReconcilingPaymentsRetrospectively'
    ]
  },
  pbg: {
    basics: [
      'nameHistory',
      'modelType',
      'goal',
      'announced',
      'performancePeriodStarts',
      'phasedIn',
      'phasedInNote'
    ],
    'general-characteristics': [
      'keyCharacteristics',
      'keyCharacteristicsNote',
      'keyCharacteristicsOther',
      'geographiesTargeted',
      'geographiesTargetedTypes',
      'geographiesTargetedAppliedTo',
      'rulemakingRequired'
    ],
    'participants-and-providers': [
      'participants',
      'medicareProviderType',
      'statesEngagement',
      'modelApplicationLevel',
      'selectionMethod',
      'providerOverlap',
      'providerOverlapHierarchy'
    ],
    beneficiaries: [
      'treatDualElligibleDifferent',
      'excludeCertainCharacteristics'
    ],
    'ops-eval-and-learning': [
      'contractorSupport',
      'contractorSupportHow',
      'anticipatedChallenges'
    ],
    payments: [
      'shouldAnyProvidersExcludedFFSSystems',
      'changesMedicarePhysicianFeeSchedule',
      'affectsMedicareSecondaryPayerClaims',
      'payModelDifferentiation',
      'creatingDependenciesBetweenServices',
      'needsClaimsDataCollection',
      'providingThirdPartyFile',
      'isContractorAwareTestDataRequirements',
      'waiverOnlyAppliesPartOfPayment',
      'beneficiaryCostSharingLevelAndHandling',
      'waiveBeneficiaryCostSharingForAnyServices',
      'nonClaimsPayments',
      'planningToUseInnovationPaymentContractor',
      'anticipateReconcilingPaymentsRetrospectively'
    ]
  },
  mdm: {
    basics: ['nameHistory'],
    beneficiaries: [
      'beneficiaries',
      'numberPeopleImpacted',
      'estimateConfidence',
      'beneficiaryOverlap',
      'precedenceRules'
    ]
  },
  cbosc: {
    basics: [
      'nameHistory',
      'announced',
      'applicationsStart',
      'performancePeriodStarts'
    ],
    'participants-and-providers': [
      'participants',
      'medicareProviderType',
      'statesEngagement',
      'expectedNumberOfParticipants',
      'estimateConfidence',
      'communicationMethod'
    ],
    'ops-eval-and-learning': [
      'stakeholders',
      'helpdeskUse',
      'contractorSupport',
      'contractorSupportHow'
    ]
  }
};

export default FilterGroupMap;
