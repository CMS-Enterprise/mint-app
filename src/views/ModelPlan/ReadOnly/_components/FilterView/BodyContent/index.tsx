import React from 'react';
import { Trans, useTranslation } from 'react-i18next';
import { Alert, Grid, Link } from '@trussworks/react-uswds';

import ReadOnlyBeneficiaries from 'views/ModelPlan/ReadOnly/Beneficiaries';
import ReadOnlyGeneralCharacteristics from 'views/ModelPlan/ReadOnly/GeneralCharacteristics';
import ReadOnlyModelBasics from 'views/ModelPlan/ReadOnly/ModelBasics';
import ReadOnlyOpsEvalAndLearning from 'views/ModelPlan/ReadOnly/OpsEvalAndLearning';
import ReadOnlyParticipantsAndProviders from 'views/ModelPlan/ReadOnly/ParticipantsAndProviders';
import ReadOnlyPayments from 'views/ModelPlan/ReadOnly/Payments';
import ReadOnlyTeamInfo from 'views/ModelPlan/ReadOnly/Team';

import allPossibleFilterViews from './_filterGroupMapping';

const FitleredViewSection = ({
  children,
  lastSection,
  sectionName
}: {
  children: React.ReactNode;
  lastSection?: boolean;
  sectionName: string;
}) => {
  return (
    <div
      className={`filtered-view-section filtered-view-section--${sectionName} margin-bottom-6 ${
        !lastSection
          ? 'border-bottom border-base-light padding-bottom-4'
          : 'padding-bottom-0'
      }`}
    >
      {children}
    </div>
  );
};

const BodyContent = ({
  modelID,
  filteredView
}: {
  modelID: string;
  filteredView: string;
}) => {
  const { t } = useTranslation('filterView');

  const individualFilterView =
    allPossibleFilterViews[filteredView as keyof typeof allPossibleFilterViews];

  return (
    <Grid>
      <FitleredViewSection sectionName="model-team">
        <h2 className="margin-top-0 margin-bottom-4">Model Team</h2>
        <ReadOnlyTeamInfo modelID={modelID} isViewingFilteredView />
      </FitleredViewSection>

      {Object.keys(individualFilterView).map(task => {
        if (task === 'basics') {
          return (
            <FitleredViewSection
              sectionName="model-basics"
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyModelBasics modelID={modelID} isViewingFilteredView />
            </FitleredViewSection>
          );
        }

        if (task === 'general-characteristics') {
          return (
            <FitleredViewSection
              sectionName={task}
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyGeneralCharacteristics
                modelID={modelID}
                isViewingFilteredView
              />
            </FitleredViewSection>
          );
        }

        if (task === 'participants-and-providers') {
          return (
            <FitleredViewSection
              sectionName={task}
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyParticipantsAndProviders
                modelID={modelID}
                isViewingFilteredView
              />
            </FitleredViewSection>
          );
        }

        if (task === 'beneficiaries') {
          return (
            <FitleredViewSection
              sectionName={task}
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyBeneficiaries modelID={modelID} isViewingFilteredView />
            </FitleredViewSection>
          );
        }

        if (task === 'ops-eval-and-learning') {
          return (
            <FitleredViewSection
              sectionName={task}
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyOpsEvalAndLearning
                modelID={modelID}
                isViewingFilteredView
              />
            </FitleredViewSection>
          );
        }

        if (task === 'payments') {
          return (
            <FitleredViewSection
              sectionName={task}
              key={task}
              lastSection={task === Object.keys(individualFilterView).pop()}
            >
              <ReadOnlyPayments modelID={modelID} isViewingFilteredView />
            </FitleredViewSection>
          );
        }
        return <></>;
      })}

      <Alert type="info" noIcon>
        <span className="margin-y-0 font-body-sm text-bold display-block">
          {t('alert.bodyContentHeading')}
        </span>
        <Trans i18nKey="filterView:alert.content">
          indexOne
          <Link href="mailto:MINTTeam@cms.hhs.gov">helpTextEmail</Link>
          indexTwo
        </Trans>
      </Alert>
    </Grid>
  );
};

export default BodyContent;
