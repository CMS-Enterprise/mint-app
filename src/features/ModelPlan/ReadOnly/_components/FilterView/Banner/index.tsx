import React from 'react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { Button, GridContainer, Icon } from '@trussworks/react-uswds';

import Tooltip from 'components/Tooltip';

import { filterGroups } from '../BodyContent/_filterGroupMapping';

import './index.scss';

type FilterViewBannerProps = {
  filteredView?: (typeof filterGroups)[number] | null;
  openFilterModal: () => void;
  openExportModal: () => void;
};

const FilterViewBanner = ({
  filteredView,
  openFilterModal,
  openExportModal
}: FilterViewBannerProps) => {
  const { t } = useTranslation('filterView');
  const { t: generalReadOnlyT } = useTranslation('generalReadOnly');

  const navigate = useNavigate();

  return (
    <div
      className="position-sticky z-100 bg-primary-darker text-white padding-105 group-filter-banner"
      data-testid="group-filter-banner"
    >
      <GridContainer>
        <div className="banner display-flex flex-justify">
          <div
            className="display-flex flex-align-center"
            style={{ gap: '1rem' }}
          >
            <Icon.Visibility size={3} aria-label="visibility" />
            <div>
              {t('youAreViewing')} <strong>{filteredView ?? t('all')}</strong>{' '}
              {filteredView !== null ? t('information') : t('allInformation')}
            </div>
            {filteredView === null && (
              <div className="mint-no-print">
                <Tooltip label={t('tooltip')} position="right">
                  <Icon.Info aria-label="info" />
                </Tooltip>
              </div>
            )}
          </div>
          <div className="mint-no-print">
            <div
              className="display-flex flex-justify flex-align-center flex-align-self-end"
              style={{ gap: '1rem' }}
            >
              {filteredView && (
                <Button
                  type="button"
                  unstyled
                  className="text-white text-no-wrap"
                  onClick={() => navigate(`${window.location.pathname}`)}
                >
                  {t('clearFilter')}
                </Button>
              )}
              <Button type="button" onClick={openFilterModal}>
                {t('filterButton')}
              </Button>

              <Button
                type="button"
                className="usa-button--outline text-white shadow-none border-white border-2px"
                onClick={openExportModal}
              >
                {generalReadOnlyT('shareExport')}
              </Button>
            </div>
          </div>
        </div>
      </GridContainer>
    </div>
  );
};

export default FilterViewBanner;
