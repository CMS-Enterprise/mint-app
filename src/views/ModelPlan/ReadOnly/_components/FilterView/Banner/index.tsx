import React, { RefObject } from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router-dom';
import {
  Button,
  GridContainer,
  IconInfo,
  IconVisiblity,
  ModalRef
} from '@trussworks/react-uswds';

import Tooltip from 'components/shared/Tooltip';
import { ShareExportModalOpener } from 'components/ShareExport/modal';

import { filterGroups } from '../BodyContent/_filterGroupMapping';

import './index.scss';

type FilterViewBannerProps = {
  filteredView?: typeof filterGroups[number];
  openFilterModal: () => void;
  shareExportModalRef?: RefObject<ModalRef>;
};

const FilterViewBanner = ({
  filteredView,
  openFilterModal,
  shareExportModalRef
}: FilterViewBannerProps) => {
  const { t } = useTranslation('filterView');
  const { t: generalReadOnlyT } = useTranslation('generalReadOnly');

  const history = useHistory();

  return (
    <div
      className="position-sticky z-100 top-0 bg-primary-darker text-white padding-105"
      data-testid="group-filter-banner"
    >
      <GridContainer>
        <div className="banner display-flex flex-justify">
          <div
            className="display-flex flex-align-center"
            style={{ gap: '1rem' }}
          >
            <IconVisiblity size={3} />
            <div>
              {t('youAreViewing')} <strong>{filteredView ?? t('all')}</strong>{' '}
              {filteredView !== null ? t('information') : t('allInformation')}
            </div>
            {filteredView === null && (
              <Tooltip label={t('tooltip')} position="right">
                <IconInfo />
              </Tooltip>
            )}
          </div>
          <div className="mint-no-print">
            <div
              className="display-flex flex-justify flex-align-center flex-align-self-end"
              style={{ gap: '1rem' }}
            >
              {filteredView && (
                <Button
                  type="button"
                  unstyled
                  className="text-white text-no-wrap"
                  onClick={() => history.push(`${history.location.pathname}`)}
                >
                  {t('clearFilter')}
                </Button>
              )}
              <Button type="button" onClick={openFilterModal}>
                {t('filterButton')}
              </Button>

              {shareExportModalRef && (
                <ShareExportModalOpener modalRef={shareExportModalRef}>
                  {generalReadOnlyT('shareExport')}
                </ShareExportModalOpener>
              )}
            </div>
          </div>
        </div>
      </GridContainer>
    </div>
  );
};

export default FilterViewBanner;
