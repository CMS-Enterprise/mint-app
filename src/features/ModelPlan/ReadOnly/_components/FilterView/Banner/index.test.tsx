import React from 'react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';
import { modelID } from 'tests/mock/readonly';

import MessageProvider from 'contexts/MessageContext';

import { filterGroups } from '../BodyContent/_filterGroupMapping';

import Banner from './index';

describe('Filter View Modal', () => {
  const openFilterModal = vi.fn();
  const openExportModal = vi.fn();
  it('renders without crashing', async () => {
    const router = createMemoryRouter(
      [
        {
          path: '/models/:modelID/collaboration-area/model-to-operations/matrix',
          element: (
            <MessageProvider>
              <Banner
                openFilterModal={openFilterModal}
                openExportModal={openExportModal}
                filteredView={'CMMI' as (typeof filterGroups)[number]}
              />
            </MessageProvider>
          )
        }
      ],
      {
        initialEntries: [
          `/models/${modelID}/collaboration-area/model-to-operations/matrix?view=solutions&hide-milestones-without-solutions=false&type=all`
        ]
      }
    );
    render(
      <MockedProvider mocks={[]} addTypename={false}>
        <RouterProvider router={router} />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(screen.getByTestId('group-filter-banner')).toBeInTheDocument();
      expect(screen.getByText('CMMI')).toBeInTheDocument();
    });
  });

  it('matches snapshot', async () => {
    const router = createMemoryRouter(
      [
        {
          path: '/models/:modelID/collaboration-area/model-to-operations/matrix',
          element: (
            <MessageProvider>
              <Banner
                openFilterModal={openFilterModal}
                openExportModal={openExportModal}
                filteredView={'CMMI' as (typeof filterGroups)[number]}
              />
            </MessageProvider>
          )
        }
      ],
      {
        initialEntries: [
          `/models/${modelID}/collaboration-area/model-to-operations/matrix?view=solutions&hide-milestones-without-solutions=false&type=all`
        ]
      }
    );

    const { asFragment } = render(
      <MockedProvider mocks={[]} addTypename={false}>
        <RouterProvider router={router} />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(screen.getByTestId('group-filter-banner')).toBeInTheDocument();
      expect(screen.getByText('CMMI')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
