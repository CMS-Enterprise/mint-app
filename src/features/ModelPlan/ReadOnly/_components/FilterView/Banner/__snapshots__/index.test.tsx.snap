// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Filter View Modal > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="position-sticky z-100 bg-primary-darker text-white padding-105 group-filter-banner"
    data-testid="group-filter-banner"
  >
    <div
      class="grid-container"
      data-testid="gridContainer"
    >
      <div
        class="banner display-flex flex-justify"
      >
        <div
          class="display-flex flex-align-center"
          style="gap: 1rem;"
        >
          <svg
            aria-label="visibility"
            class="usa-icon usa-icon--size-3"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <div>
            You are viewing 
            <strong>
              CMMI
            </strong>
             information.
          </div>
        </div>
        <div
          class="mint-no-print"
        >
          <div
            class="display-flex flex-justify flex-align-center flex-align-self-end"
            style="gap: 1rem;"
          >
            <button
              class="usa-button usa-button--unstyled text-white text-no-wrap"
              data-testid="button"
              type="button"
            >
              Clear filter
            </button>
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              Filter
            </button>
            <button
              class="usa-button usa-button--outline text-white shadow-none border-white border-2px"
              data-testid="button"
              type="button"
            >
              Share or export
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
